Grammar/python.gram

```diff
@@ -126,6 +126,7 @@ simple_stmt[stmt_ty] (memo):
     | &'nonlocal' nonlocal_stmt
 
 compound_stmt[stmt_ty]:
+    | invalid_compound_stmt
```

Comment:
This is more acceptable, but it will trigger with any compound statement:

```
with A():
    elif 3:
        ...
SyntaxError: 'elif' must match an if-statement here
```

Also, a slight minor problem is that at this level we are catching an incorrect keyword in a specific bad context, but there could be lot more. For instance, if you would write:

```
[x for x in range(3) if 34 elif 12]
```

Reply:
Is there anything I should do about that?

Reply:
No for now, let's discuss first if we are ok with this. We may need to adapt the error message

Reply:
> No for now, let's discuss first if we are ok with this.

I mean, let's wait to see what @lysnikolaou and @isidentical think

Reply:
Maybe also @cfbolz

Reply:
Should we keep waiting for the others to review?

Reply:
@pablogsal should we keep waiting for the others to review?

Reply:
@pablogsal should we keep waiting for the other reviewers, or should this pull request be closed?

Reply:
> @pablogsal should we keep waiting for the other reviewers, or should this pull request be closed?

We should wait for other reviewers.

Reply:
I am currently unable to comment on this issue, so feel free to skip me.

Reply:
It's been 7 days and I plan on closing this pull request because of inactivity. Should we keep waiting for @lysnikolaou?

Reply:
@pablogsal It's been 7 days and I plan on closing this pull request because of inactivity. Should we keep waiting for @lysnikolaou?